From 06d713fefaf19cb4dfb83126ba1c078d6dcabc67 Mon Sep 17 00:00:00 2001
From: xbmc <fernetmenta@online.de>
Date: Tue, 25 Dec 2012 08:35:36 +0100
Subject: [PATCH] ffmpeg: backport latm: fix initialization on some streams
 when no extradata is available:
 d039b6074ca68da9b6dc88d8bb40056fee9fecb6

---
 libavcodec/aacdec.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libavcodec/aacdec.c b/libavcodec/aacdec.c
index 5c6404e..ad2991b 100644
--- a/libavcodec/aacdec.c
+++ b/libavcodec/aacdec.c
@@ -2403,7 +2403,8 @@ static int latm_decode_audio_specific_config(struct LATMContext *latmctx,
     if (bits_consumed < 0)
         return AVERROR_INVALIDDATA;
 
-    if (ac->m4ac.sample_rate != m4ac.sample_rate ||
+    if (!latmctx->initialized ||
+        ac->m4ac.sample_rate != m4ac.sample_rate ||
         ac->m4ac.chan_config != m4ac.chan_config) {
 
         av_log(avctx, AV_LOG_INFO, "audio config changed\n");
-- 
1.7.10

